### ISSUE TITLE
Using `store.setState` instead of the provided `set` function

### ISSUE THREAD

**Issue by acnebs** [charkour/zundo#67]
Hi, trying out your library and like the look of it so far! One issue I'm having though is that typically when using Zustand I'm not a huge fan of defining my setter functions in advance, and instead like using `store.setState`, e.g.

```tsx
const store = create(() => ({
  bears: 0,
}))

const BearComponent = () => (
  <div>
    <button onClick={() => store.setState({ bears: 10 })}>Set bears to 10</button>
  </div>
)
```

But when using zundo, this doesn't seem to work, and instead you need to use `set`, e.g.

```tsx
const store = create()(temporal(set => ({
  bears: 0,
  setBears: (bears) => set({bears}),
})))
```

I've also made a [codesandbox here](https://codesandbox.io/s/zundo-but-using-setstate-qiwlwh).
Is this supposed to work? Is this a bug in the 2.0 beta? Thanks!

**Comment by charkour** [comment]
Hi @acnebs!

Thanks for the bug report with reproduction. 

This wasn't a use case I accounted for when writing this, so it is a feature that will need to be added. 

I'll be able to take a look at this this weekend, but if you want to take a look at it sooner, we need to do the same pattern with the modified setter and modify the setState function. I'm not sure of hand how easy that is to update, but I'll try my best. 

Thanks

**Comment by charkour** [comment]
A work around for the time being is to define setters inside the store rather than inline.

**Comment by charkour** [comment]
I've figured out a solution and will make a PR soon.

**Comment by charkour** [comment]
This has been fixed in v2.0.0-beta.10

Thank you for reporting!

Here is the [updated code sandbox](https://codesandbox.io/s/zundo-but-using-setstate-forked-k5yex7)

**Comment by acnebs** [comment]
Thanks @charkour, that looks great and is exactly what I was looking for!
